tariff/pun.go

```diff
@@ -123,48 +127,67 @@ func (t *Pun) Type() api.TariffType {
 }
 
 func (t *Pun) getData(day time.Time) (api.Rates, error) {
-	// Cookie Jar zur Speicherung von Cookies zwischen den Requests
+	// Cookie Jar to store cookies between requests
 	client := request.NewClient(t.log)
 	client.Jar, _ = cookiejar.New(nil)
 
-	// Erster Request
-	uri := "https://storico.mercatoelettrico.org/It/WebServerDataStore/MGP_Prezzi/" + day.Format("20060102") + "MGPPrezzi.xml"
-	resp, err := client.Get(uri)
+	// Request the ZIP file
+	uri := "https://gme.mercatoelettrico.org/DesktopModules/GmeDownload/API/ExcelDownload/downloadzipfile?DataInizio=" + day.Format("20060102") + "&DataFine=" + day.Format("20060102") + "&Date=" + day.Format("20060102") + "&Mercato=MGP&Settore=Prezzi&FiltroDate=InizioFine"
+	req, err := http.NewRequest("GET", uri, nil)
 	if err != nil {
 		return nil, err
 	}
+
+	req.Header = http.Header{
+		"moduleid":           []string{"12103"},
+		"Referer":            []string{"https://gme.mercatoelettrico.org/en-us/Home/Results/Electricity/MGP/Download?valore=Prezzi"},
+		"sec-ch-ua-mobile":   []string{"?0"},
+		"sec-ch-ua-platform": []string{"Windows"},
+		"sec-fetch-dest":     []string{"empty"},
+		"sec-fetch-mode":     []string{"cors"},
+		"sec-fetch-site":     []string{"same-origin"},
+		"sec-gpc":            []string{"1"},
+		"tabid":              []string{"1749"},
+		"userid":             []string{"-1"},
+	}
+
+	resp, err := client.Do(req)
+	if err != nil || resp.StatusCode != http.StatusOK {
```

Comment:
Status check is already done by client

Reply:
tried, without the status check if the data is not available for the next day (until 13 AM of the current day) and the server replies with 404, the whole tariff is not loaded